services/QuillLMS/lib/tasks/schools.rake

```diff
@@ -107,4 +107,61 @@ namespace :schools do
       end
     end
   end
+
+  # A bit more detail here: when we did our original population of school
+  # data from federal NCES data sources, NCES ID values included left-
+  # padding "0"s to make them all the same length.  When we updated our
+  # data in 2022 from the same federal sources, the NCES IDs were no longer
+  # left-padded in the data set, so we ended up creating duplicate schools.
+  # One set with leading "0"s and one without.  This task is intended to
+  # move any users from the duplicates to the original schools, and then to
+  # remove the duplicates from our database.
+  desc 'Clean up duplicate schools based on 0-left-padded NCES ID'
+  task :clean_up_duplicates => :environment do
+
+    teachers_assigned_to_duplicates = <<-SQL
+      SELECT original.id AS original_school_id,
+            duplicate.id AS duplicate_school_id,
+            schools_users.id AS schools_users_id,
+            schools_admins.id AS schools_admins_id
+          FROM schools AS original
+          JOIN schools AS duplicate
+              ON original.nces_id = CONCAT('0', duplicate.nces_id)
```

Comment:
Lines 129 and 130 have identical expressions. 

Furthermore, you should be able to accomplish the original-to-duplicate matching without an OR statement, since each side of the join will have , for example, id `012` and id `12`.

Reply:
Also, with this query

`select count(*) from schools where nces_id::varchar(255) LIKE '0%'` 

we observe that only 19k, or ~12% of all schools, have an `nces_id` with a leading zero. Adding this filter to your query should cause a significant speed up.

Reply:
@happythenewsad Can you expand on what you mean in this second comment?  Are you suggesting adding a `WHERE` clause to the overall query, or are you suggesting something else?

Reply:
Up to you... I won't hold up the PR because of this, since the rake task may complete in a timely manner without it.